src/mongocxx/collection.cpp

```diff
@@ -65,6 +65,23 @@ stdx::string_view collection::name() const noexcept {
     return _impl->name;
 }
 
+void collection::name(stdx::string_view new_name, bool overwrite_old) {
+  bson_error_t error;
+
+  _impl->name = (std::string)new_name;
+  auto result = mongoc_collection_rename(
+      _impl->collection_t
+    , _impl->database_name.c_str()
+    , _impl->name.c_str()
```

Comment:
Interesting... I suggested it because it is used elsewhere (ex: [collection constructor](https://github.com/mongodb/mongo-cxx-driver/blob/master/src/mongocxx/collection.cpp#L70)).
`c_str` is probably safer though.